src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -1584,16 +1549,21 @@ impl OrchestratorWorker {
                 .build()
                 .context("error building HTTP client")?;
             let resp = http_client.get(metrics_url).send().await?;
-            let usage: Usage = resp.json().await?;
-
-            usage
-                .disk_bytes
-                .ok_or_else(|| anyhow!("process did not provide disk usage"))
+            let Usage {
+                disk_bytes,
+                swap_bytes,
+            } = resp.json().await?;
+
+            let bytes = match (disk_bytes, swap_bytes) {
+                (Some(disk), Some(swap)) => Some(disk + swap),
+                (Some(bytes), None) | (None, Some(bytes)) => Some(bytes),
+                (None, None) => None,
+            };
```

Comment:
```suggestion
            let bytes = disk_bytes.into_iter().chain(swap_bytes).fold(None, |r, i| Some(r.unwrap_or(0) + i))
```
Maybe?

Reply:
I find the match clearer! It is a bit verbose though, I'll replace it with

```
if let (Some(disk), Some(swap)) = (disk_bytes, swap_bytes) {
    Some(disk + swap)
} else {
    a.or(b)
}
```

It would sure be nice if there was an `Option` combinator with the same semantics...